experiment/nipopow_verifier/tools/proof/proof.py

```diff
@@ -75,3 +89,216 @@ def set(self, proof, proof_name=""):
             self.hashed_headers.append(Hash(self.proof[i][0]))
 
         self.size = len(self.proof)
+
+        (
+            self.best_level,
+            self.best_score,
+            self.levels,
+            self.scores,
+        ) = best_level_and_score(self.proof)
+
+        for level in sorted(self.levels.keys()):
+            print(
+                "Level", level, "has", self.levels[level], "blocks with score", end=""
+            )
+            if level == self.best_level:
+                print(" " + str(self.best_score) + " <- best")
+            else:
+                print(" " + str(self.scores[level]))
+
+        if header is None or header_map is None or interlink_map is None:
+            print("No interlink provided")
+            return
+        else:
+            print(
+                "Interlink provided. Creating isolated proof at level", self.best_level
+            )
+
+        self.best_level_subproof = isolate_proof_level(
+            self.best_level, proof, header, header_map, interlink_map
+        )
+        (
+            self.best_level_subproof_headers,
+            self.best_level_subproof_siblings,
+        ) = self.extract_headers_siblings(self.best_level_subproof)
+        self.best_level_subproof_size = len(self.best_level_subproof)
+
+
+def header_level(header):
+    """
+    Computes the level of a proof block
+    """
+
+    n_bits = header[104:108]
+    n_bits_int = int().from_bytes(n_bits, "little")
+    target = blockchain_utils.bits_to_target(n_bits_int)
+    header_hash = blockchain_utils.uint256_from_str(blockchain_utils.Hash(header))
+    level = (int(target / header_hash)).bit_length() - 1
+    return level
+
+
+# Linter made me do it
+def best_level_and_score(proof, miou=6):
```

Comment:
made you do what?

Reply:
miou instead of m

Reply:
miou read in greek is μ, which this is not. Mask the linter here.

Reply:
Masked